### PR TITLE
Rsc create staffing companies 

### PR DESCRIPTION
**Pull request by RhonGabriel**
# PT
https://www.pivotaltracker.com/story/show/161592665

-Added feature test with sad path
- Added email column for staffing company model
- Added new, create and show action
- Create form for adding company information
- Rendered page using js in order to show forms in modals/pop-up
- Added validation of email format and uniqueness

screenshot 2018-11-06 at 11 15 31

### CODE REVIEW

**Review by tochman — COMMENTED** [review]
Please look through Zak’s PR with focus on my last commits. I would also advise you to wait until that PR is merged.

File: app/controllers/staffcos_controller.rb
Comment (tochman) [inline app/controllers/staffcos_controller.rb]: Use singular form please

File: app/controllers/staffcos_controller.rb
Comment (tochman) [inline app/controllers/staffcos_controller.rb:16]: What if the object is NOT persisted? You need a `else` block to handle the exeption

File: app/controllers/staffcos_controller.rb
Comment (tochman) [inline app/controllers/staffcos_controller.rb]: Code formatting is off

File: app/models/staffco.rb
Comment (tochman) [inline app/models/staffco.rb:3]: Good job. Haven't seen this before.

File: app/views/staffcos/_new.html.haml
Comment (tochman) [inline app/views/staffcos/_new.html.haml]: You probably want to indent the form to be within `#list-item.flex` But, on a second thought, why do you need the id `list-item`?

Comment (tochman) [reply app/views/staffcos/_new.html.haml]: You probably also need a div to display error messages if the controller sends out any. I added a div with id `error_messages` in the other PR

File: app/views/staffcos/_new.html.haml
Comment (tochman) [inline app/views/staffcos/_new.html.haml]: DO NOT wrap a link in a `button` tag. That is not the way HTML works and even if your browser render it, you are bound to get an ambiguous match when you run your tests.

File: app/views/staffcos/_new.html.haml
Comment (tochman) [inline app/views/staffcos/_new.html.haml]: You probably need a Cancel button on this pop-uo

File: app/views/staffcos/new.js.erb
Comment (tochman) [inline app/views/staffcos/new.js.erb:5]: You are listening for a click on a Cancel button, but you are not displaying one

File: app/views/staffcos/new.js.erb
Comment (tochman) [inline app/views/staffcos/new.js.erb:10]: This element is missing from your form and pop-up

File: features/rsc_can_create_new_staffing_companies.feature
Comment (tochman) [inline features/rsc_can_create_new_staffing_companies.feature]: I would like you to test for the success message/flash

File: features/rsc_can_create_new_staffing_companies.feature
Comment (tochman) [inline features/rsc_can_create_new_staffing_companies.feature]: I would like you to test for the error message/flash

**Review by tochman — COMMENTED** [review]
You are getting there but I need you to carefully read every line of code that you copy from other parts of the application. The pattern of rendering forms and pages using javascript is a bit complicated and needs your full attention.

File: app/controllers/staffcos_controller.rb
Comment (tochman) [inline app/controllers/staffcos_controller.rb]: Sorry, here you can use plural format. It’s a collection

File: app/views/staffcos/_index.html.haml
Comment (tochman) [inline app/views/staffcos/_index.html.haml]: Same here. Plural

File: app/controllers/staffcos_controller.rb
Comment (angiecstn) [inline app/controllers/staffcos_controller.rb]: Is the show action needed?

File: features/step_definitions/assertion_steps.rb
Comment (tochman) [inline features/step_definitions/assertion_steps.rb]: Why use parathesis around the variable?

File: features/step_definitions/assertion_steps.rb
Comment (tochman) [inline features/step_definitions/assertion_steps.rb]: Wouldn't `contact_name` be a more suitable attribute name?

Comment (RhonGabriel) [reply features/step_definitions/assertion_steps.rb]: changed

**Review by Noelryn — APPROVED** [review]

**Review by angiecstn — APPROVED** [review]

File: features/rsc_can_create_new_staffing_companies.feature
Comment (tochman) [inline features/rsc_can_create_new_staffing_companies.feature:18]: I don't mind the abstraction. It's good. But IMO you should be more in control of the name of the company toy are creating, so that you can add an assertion at the end of the scenario to make sure that the object has been created AND is being shown on the index.

**Review by Noelryn — APPROVED** [review]

**Review by angiecstn — APPROVED** [review]

**Pull request merged**
